crates/brainfuck_prover/src/brainfuck_air/mod.rs

```diff
@@ -184,10 +199,9 @@ pub fn verify_brainfuck(
     let interaction_elements = BrainfuckInteractionElements::draw(channel);
     // Check that the lookup sum is valid, otherwise throw
     // TODO: panic! should be replaced by custom error
-    assert!(
-        !lookup_sum_valid(&claim, &interaction_elements, &interaction_claim),
-        "Invalid lookup sum."
-    );
+    if !lookup_sum_valid(&claim, &interaction_elements, &interaction_claim) {
+        return Err(BrainfuckVerificationError::InvalidLogupSum);
```

Comment:
Directly use the InvalidLookup variant from VerificationError?

```suggestion
        return Err(VerificationError::InvalidLookup("Invalid logup sum".to_string()));
```